tiledb/ml/models/base.py

```diff
@@ -9,13 +15,125 @@ class TileDBModel(abc.ABC):
     store machine learning models (Tensorflow, PyTorch, ect) as TileDB arrays.
     """
 
-    def __init__(self, uri: str):
+    def __init__(self, uri: str, ctx=None, namespace=None):
         """
         Base class for saving machine learning models as TileDB arrays
         and loading machine learning models from TileDB arrays.
         :param uri: str
         """
         self.uri = uri
+        self.tiledb_uri = None
```

Comment:
Agreed. `uri`, and `tiledb_uri` though are different as per the discussion bellow. 
`tiledb_uri` is calculated and does not replace `uri`

```
self.uri = "tiledb://{}/{}".format(
                namespace, os.path.join(s3_prefix, uri)
            )
```

modifies `uri` with the s3 notation if user's profile has a bucket related known to store files of supported types (`notebook`, `registered_udfs`, `ml_models`, ...)

More changes will be needed, for example what happens if user adds a full local path in `uri` instead of saving in current path as most examples here

Reply:
Ah, I see what you mean was not sure if `self.x` is visible from the class when declared first time in `if`
removed first time assignments